subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,170 @@
+            artifactTransform(FileSizer)
+        }
+    }
+}
+
+import javax.inject.Inject
+
+class FileSizer extends ArtifactTransform {
+
+    FileCollection artifactDependencies
+
+    @Inject
+    FileSizer(FileCollection artifactDependencies) {
+        this.artifactDependencies = artifactDependencies
+        println "Creating FileSizer"
+    }
+    
+    List<File> transform(File input) {
+        println "Received dependencies files \${artifactDependencies.files*.name} for processing \${input.name}"
+
+        assert outputDirectory.directory && outputDirectory.list().length == 0
+        def output = new File(outputDirectory, input.name + ".txt")
+        println "Transforming \${input.name} to \${output.name}"
+        output.text = String.valueOf(input.length())
+        return [output]
+    }
+}
+
+"""
+    }
+
+    def "transform can access artifact dependencies as FileCollection when using ArtifactView"() {
+
+        given:
+
+        buildFile << """
+project(':app') {
+    task resolve(type: Copy) {
+        def artifacts = configurations.compileClasspath.incoming.artifactView {
+            attributes { it.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.class, 'size')) }
+        }.artifacts
+        from artifacts.artifactFiles
+        into "\${buildDir}/libs"
+        doLast {
+            println "files: " + artifacts.collect { it.file.name }
+            println "ids: " + artifacts.collect { it.id }
+            println "components: " + artifacts.collect { it.id.componentIdentifier }
+            println "variants: " + artifacts.collect { it.variant.attributes }
+        }
+    }
+}
+
+"""
+
+        when:
+        run "resolve"
+
+        then:
+        output.count('Transforming') == 4
+        output.contains('Received dependencies files [slf4j-api-1.7.25.jar] for processing lib.jar')
+        output.contains('Received dependencies files [hamcrest-core-1.3.jar] for processing junit-4.11.jar')
+    }
+
+    def "transform can access artifact dependencies as FileCollection when using configuration attributes"() {
+
+        given:
+
+        buildFile << """
+project(':app') {
+    configurations {
+        sizeConf {
+            attributes.attribute(artifactType, 'size')
+            extendsFrom compileClasspath
+        }
+    }
+
+    task resolve(type: Copy) {
+        def artifacts = configurations.sizeConf.incoming.artifacts
+        from artifacts.artifactFiles
+        into "\${buildDir}/libs"
+        doLast {
+            println "files: " + artifacts.collect { it.file.name }
+            println "ids: " + artifacts.collect { it.id }
+            println "components: " + artifacts.collect { it.id.componentIdentifier }
+            println "variants: " + artifacts.collect { it.variant.attributes }
+        }
+    }
+}
+"""
+
+        when:
+        run "resolve"
+
+        then:
+        output.count("Transforming") == 4
+        // The asserts below should not have the .txt part
```

Comment:
Shall this be marked as `TODO`? Also, we should probably use `@ToBeImplemented` on this one.

Reply:
Done for the `@ToBeImplemented`